src/dbnode/persist/fs/files.go

```diff
@@ -832,18 +925,9 @@ func FileSetAt(filePathPrefix string, namespace ident.ID, shard uint32, blockSta
 		return FileSetFile{}, false, err
 	}
 
-	matched.sortByTimeAscending()
-	for i, fileset := range matched {
-		if fileset.ID.BlockStart.Equal(blockStart) {
-			nextIdx := i + 1
-			if nextIdx < len(matched) && matched[nextIdx].ID.BlockStart.Equal(blockStart) {
-				// Should never happen
-				return FileSetFile{}, false, fmt.Errorf(
-					"found multiple fileset files for blockStart: %d",
-					blockStart.Unix(),
-				)
-			}
-
+	matched.sortByTimeAndVolumeIndexAscending()
+	for _, fileset := range matched {
+		if fileset.ID.BlockStart.Equal(blockStart) && fileset.ID.VolumeIndex == volume {
```

Comment:
We removed the check for seeing if the fileset that comes after has the same block start + volume, do we want to keep it perhaps?

Reply:
Sure.